src/State/LocaleDetector.php

```diff
@@ -105,9 +105,10 @@ public function detectBrowserLocale(HTTPRequest $request, $currentDomain = false
         // Check each requested locale against loaded locales
         foreach ($prioritisedLocales as $priority => $parsedLocales) {
             foreach ($parsedLocales as $browserLocale) {
-                foreach ($this->getLocales($currentDomain) as $locale) {
-                    if (stripos(preg_replace('/_/', '-', $locale), $browserLocale) === 0) {
-                        return $locale;
+                foreach ($this->getLocales($currentDomain) as $localeObj) {
+                    /** @var Locale $localeObj */
+                    if (stripos(preg_replace('/_/', '-', $localeObj->Locale), $browserLocale) === 0) {
```

Comment:
Can we put this preg_replace into a separate getter on Locale class? (After I merge this)